crates/deltalake-core/src/operations/write.rs

```diff
@@ -108,6 +108,8 @@ pub struct WriteBuilder {
     write_batch_size: Option<usize>,
     /// RecordBatches to be written into the table
     batches: Option<Vec<RecordBatch>>,
+    /// whether to overwrite the schema
+    overwrite_schema: bool,
```

Comment:
is this meant for schema evolution? If so, I'd recommend moving that to a follow-up PR as it would likely blow up this PR quite a bit.

Reply:
+1. I think it's fine if we let that return `NotImplementedError` for now.

Reply:
It was an quick attempt for schema evolution. I was able to write except it didn't write the columns that were not part of the original schema, so I need to dig through the code more.

Ok, let's do this as improvement in another update.

Reply:
we also would need to update all read path to always add null columns for columns non-existent in older parquet files. Haven't looked into it, but this would likely require some larger refactoring particularly in the datafusion `DeltaScan`. Saying this we likely need to validate that added columns are always nullable as well.

Reply:
I think that would be schema evolution for also appends.

The PyArrow writer can do schema evolution but only combined with an `overwrite` mode. 

I think that's purely a metadata action then. Would this be doable with the existing deltalake-core crate?

Reply:
not sure, we may end up with unreadble tables if we do this... if we replace the whole table this might work.

Reply:
With PyArrow it only works together with overwrite so it should be safe. Is there a way to adjust the commit that's written during a write?